### ISSUE TITLE
Namespaces taking abnormally long time stuck in Activating state resulting in UI timing out and errors when make first workload

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#20079]
Version: v2.2.3-rc6, v2.2.3-rc7 (and possibly additional previous v2.2.3 rc's)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create any new RKE cluster (in my case DO, EC2, Azure all tested)
- Create a project "workload" and deploy your first workload there (when creating this first workload it should also be creating a new namespace). You must have never created a ns / workload on this new cluster before.

**Result:**
In the UI you will get an error because the API is leaving the Namespace in the Activating state for an abnormally long time (generally under 15 seconds but still takes a long time compared to previous releases, etc). The "Saving..." button spins for quite a while.
![image](https://user-images.githubusercontent.com/45179589/57410975-240cc200-71a1-11e9-9e85-ef1cc4619422.png)
![image](https://user-images.githubusercontent.com/45179589/57410987-2a02a300-71a1-11e9-9d46-4b25eabdd0b4.png)

In the UI if you try to launch your workload again you will get a null error. This is because the UI is trying to create the namespace again because it took too long to go into active state.
![image](https://user-images.githubusercontent.com/45179589/57411005-2ff88400-71a1-11e9-8708-27c0377fcef8.png)

If you cancel out and view the Namespaces page you'll see in the UI it claims to still be activating forever (even if eventually becomes Active)
![image](https://user-images.githubusercontent.com/45179589/57411064-5dddc880-71a1-11e9-983c-f77875dce150.png)

You have to manually refresh the page in order to see it switch to "Active" state.

If you create a second workload of any time it should work fine. Namespaces get created faster now and we don't run into the problem again on this cluster.

**Other details that may be helpful:**
No errors/warnings in rancher logs at INFO level.

I had a hard time reproducing this issue on the latest master, thus I assume this issue is in v2.2.3 only. I have not observed the problem while testing v2.2.2 but did not specifically re-test it recently. I doubt this is an issue in v2.2.2.

**Environment information**
HA or single-node rancher both affected

**Comment by alena1108** [comment]
@davidnuzik can you share the output of `kubectl get ns -o json output` for the ns stuck in activating

@orangedeng most likely one of the conditions is not set, can you investigate

**Comment by orangedeng** [comment]
I can not reproduce this issue in master 5/10 with EC2 cluster. From the issue description, the ns we create never gets status updated. It seems like a problem that UI didn't get the ns status update via websocket.

**Comment by alena1108** [comment]
Based on offline talk with @davidnuzik , the issue has not been observed lately. Marking as to-test in v2.2.4-rc2

**Comment by davidnuzik** [comment]
Versions:
v2.2.4-rc4 single-node
v2.2.4-rc4 HA setup

I am no longer observing the issue I originally reported. I am not certain as to why this is no longer occurring, but I tested carefully with each setup (HA and non-HA) again with all the different providers I reported. When I originally reported the issue this was easily reproducible and in multiple environments including ones I didn't create myself.

Since I am no longer observing this behavior I am closing the issue.
